src/cmap/connection.ts

```diff
@@ -488,18 +488,19 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
     options: CommandOptions | undefined,
     callback: Callback
   ): void {
-    const readPreference = getReadPreference(cmd, options);
+    cmd = { ...cmd };
```

Comment:
Although this leaves the supplied command untouched outside of the scope of this method can we assign it to a new const to avoid any potential confusion?